sdk/generic-sdk-core/generic-core-http-okhttp/src/main/java/com/generic/core/http/okhttp/OkHttpHttpClient.java

```diff
@@ -138,40 +141,74 @@ private static long getRequestContentLength(BinaryData content, HttpHeaders head
         return contentLength;
     }
 
-    private Response<?> toResponse(HttpRequest request, okhttp3.Response response, boolean eagerlyReadResponse,
-                                   boolean ignoreResponseBody, boolean eagerlyConvertHeaders) throws IOException {
+    private Response<?> toResponse(HttpRequest request, okhttp3.Response response,
+                                   ResponseBodyMode responseBodyMode, boolean eagerlyConvertHeaders) throws IOException {
         okhttp3.Headers responseHeaders = response.headers();
 
         if (isTextEventStream(responseHeaders) && response.body() != null) {
             ServerSentEventListener listener = request.getServerSentEventListener();
+
             if (listener != null) {
-                processTextEventStream(request,
-                    httpRequest -> this.send(httpRequest), response.body().byteStream(), listener, LOGGER);
+                processTextEventStream(request, this::send, response.body().byteStream(), listener, LOGGER);
             } else {
                 throw LOGGER.logThrowableAsError(new RuntimeException(ServerSentEventUtil.NO_LISTENER_ERROR_MESSAGE));
             }
-            return new OkHttpResponse(response, request, eagerlyConvertHeaders, EMPTY_BODY);
+
+            return new OkHttpResponse(response, request, eagerlyConvertHeaders, BinaryData.fromBytes(EMPTY_BODY));
         }
-        return processResponse(request, response, eagerlyReadResponse, ignoreResponseBody, eagerlyConvertHeaders);
+
+        return processResponse(request, response, responseBodyMode, eagerlyConvertHeaders);
     }
 
-    private Response<?> processResponse(HttpRequest request, okhttp3.Response response, boolean eagerlyReadResponse,
-                                        boolean ignoreResponseBody, boolean eagerlyConvertHeaders) throws IOException {
-        if (eagerlyReadResponse || ignoreResponseBody) {
-            try (ResponseBody body = response.body()) {
-                byte[] bytes = (body != null) ? body.bytes() : EMPTY_BODY;
-                return new OkHttpResponse(response, request, eagerlyConvertHeaders, bytes);
+    private Response<?> processResponse(HttpRequest request, okhttp3.Response response,
+                                        ResponseBodyMode responseBodyMode, boolean eagerlyConvertHeaders) throws IOException {
+        if (responseBodyMode == null) {
```

Comment:
Should we log in the cases the specified responseBodyMode doesn't work with the Content-Type of the response body?
Or do we try to do best-effort conversions in that case or preference is given to the responseBodyMode over ContentType?

Reply:
Currently, we're giving preference to the mode specified by the user. If they don't specify anything we go by what we've determined to be the default based on operation/response type.

Reply:
I don't think that can happen. Do you have an example in mind?

Reply:
For example, user has set BUFFER for SSE response, what would that do?
cc: @srnagar

Reply:
SSE responses are unaffected by `ResponseHandlingMode` for the time being. We can change that in a future PR.